### ISSUE TITLE
[Bug]: shopify theme dev error

### ISSUE THREAD

**Issue by brady-sewall** [Shopify/cli#1425]
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

Theme

### Expected behavior

The Shopify CLI to start up and display links to the dev store.

### Actual behavior

`bundler: not executable: /opt/homebrew/Cellar/shopify-cli/3.44.0/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/bin/shopify`
Error
```
    at execCLI2 (file:///opt/homebrew/Cellar/shopify-cli/3.44.0/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/dist/public/node/ruby.js:55:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

### Reproduction steps

1. Go to theme directory on local machine
2. Run 'shopify theme dev --store [myshopifystore]'

Shopify CLI was working fine yesterday. Today Homebrew noted that Shopify-CLI 3.44.0 was available and (I think) a Node update as well. After updating via `brew upgrade`, Shopify-CLI wasn't working.

### Operating System

Mac OS Ventura 13.2.1

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.44.0

### Shell

Hyper and VSCode

### Node version (run `node -v` if you're not sure)

v19.7.0

### What language and version are you using in your application?

Liquid, HTML, CSS, JS

**Comment by petercampanelli** [comment]
Having exactly same issue after running `brew upgrade`

**Comment by ouw-jvt** [comment]
+1 unable to run theme dev as of the most recent update

**Comment by kenji-twp** [comment]
Same issue here after upgrading to 3.44.0

**Comment by Arkham** [comment]
Thanks for the reports folks, most of the team is EMEA based so we're going to release a fix tomorrow for homebrew installations. Another fix I've found is to make the binary executable like so:
```
chmod a+x /opt/homebrew/Cellar/shopify-cli/3.44.0/libexec/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/assets/cli-ruby/bin/shopify
```
After that I was able to run `shopify theme` commands. Does this fix things for you as well?

**Comment by jtleathers** [comment]
@Arkham That definitely worked for me. Thanks!

**Comment by phv-ppuente** [comment]
As mentioned in https://github.com/Shopify/cli/issues/1425#issuecomment-1440957987 I also get this error when running `shopify theme push` from Github actions. Prepending the command with `SHOPIFY_CLI_BUNDLED_THEME_CLI=1` appears to be a valid workaround.

**Comment by mongorian346** [comment]
I still have this problem.
Every time We have to commend with SHOPIFY_CLI_BUNDLED_THEME_CLI=1？

<img width="812" alt="スクリーンショット 2023-02-24 12 15 25" src="https://user-images.githubusercontent.com/60023453/221083358-eccd4656-b688-461e-af7b-6c237f0c0a7c.png">

**Comment by kyrylo-soulandwolf** [comment]
Also, still getting this error after 3.44.*

```
Your user account isn't allowed to install to the system RubyGems.
  You can cancel this installation and run:

      bundle config set --local path 'vendor/bundle'
      bundle install

  to install the gems into ./vendor/bundle/, or you can enter your password
  and install the bundled gems to RubyGems using sudo.
```

**Comment by gonzaloriestra** [comment]
@mongorian346 could you try upgrading to v3.44.1?

@kyrylo-soulandwolf that looks like a different issue. Could you run the command with `--verbose` and share the log?

**Comment by amcaplan** [comment]
Interesting, I think this is the first time I've heard of the CLI for themes being installed in the project. Cool piece of data for us!

At the same time, it isn't the usual way to install it, so there may be some unexpected issues... but I think we should support it nonetheless.

I think https://github.com/Shopify/cli/pull/1433 should fix it.

**Comment by truleighsyd** [comment]
I'm also experiencing the same issue I've installed CLI3 using brew and npm -g. Both are running the same error. The current version Shopify CLI is 3.44.1

`Error
    at execCLI2 (file:///usr/local/lib/node_modules/@shopify/theme/node_modules/@shopify/cli-kit/dist/public/node/ruby.js:55:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)`

The above recommended solutions did not work.

**Comment by kyrylo-soulandwolf** [comment]
@amcaplan Thanks for looking into this!
I've explained our use case here https://github.com/Shopify/cli/issues/1320#issuecomment-1424042652 (this particular issues comes and goes between the versions) and I think it should definitely be supported :)

**Comment by kyrylo-soulandwolf** [comment]
@amcaplan hey there. Just coming back to this issue to find out about a rough ETA on a new CLI version with your mentioned fix (https://github.com/Shopify/cli/pull/1433)? It's merged into master but there hasn't been a new release in more than a month :cry:

**Comment by Arkham** [comment]
hi @kyrylo-soulandwolf, we just released `3.45.0` which includes the above fix. In the future, you can opt in prerelease versions which come out weekly:
- for apps: https://shopify.dev/docs/apps/tools/cli#pre-release-cli-versions
- for themes: https://shopify.dev/docs/themes/tools/cli/install#pre-release-cli-versions

**Comment by syedsimanta03** [comment]
It's been a long time shopify can't fix CLI issues. The root is they are using ruby
